packages/xsnap/test/xsnap.test.js

```diff
@@ -522,6 +523,34 @@ test('GC after snapshot vs restore', async t => {
   t.is(workerGC, cloneGC);
 });
 
+test.failing('isolate segfault', async t => {
+  const xsnapr = pickXSnap();
+
+  const opts = { ...options(io), name: 'original', meteringLimit: 0 };
+  const worker = await xsnapr(opts);
+  t.teardown(worker.terminate);
+
+  await worker.evaluate(`
+  const runToGC = async () => {
+    let collected = false;
+    const fr = new FinalizationRegistry(() => {
+      collected = true;
+    });
+    fr.register({}, undefined);
+    const trashCan = [];
+
+    let qty;
+    for (qty = 0; !collected; qty += 1) {
+      await new Promise(setImmediate);
+      trashCan.push(Array(10_000).map(() => ({})));
+    }
+  };
+  runToGC();
+
+  Array.from(Array(1_047_850).keys()).length;
+  `);
+});
```

Comment:
This is a subset of the above test that exhibits the failure mode. Again, the last figure is right at the edge of pass/fail.